### PR TITLE
Fix sanitize column names modifying the data frame in distributed mode

### PR DESCRIPTION
**Pull request by LeonLuttenberger**
### Feature or Bugfix
- Bugfix

### Detail
- Fix sanitize column names modifying the data frame in distributed mode

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 7378ac1367bcbb601898512e7065d9e064d05929
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=80dc284d-d28a-45e3-ba31-3b34c768590d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubStandardCodeBuild8C06-llutOAimTATs
* Commit ID: 7378ac1367bcbb601898512e7065d9e064d05929
* Result: SUCCEEDED
* [Build Logs](https://jsv9u8gote.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=beca2e6f-404e-48b8-902f-bbdc439c9e1a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-ATYtnXPE7MOa
* Commit ID: 7378ac1367bcbb601898512e7065d9e064d05929
* Result: FAILED
* [Build Logs](https://i1emq028pj.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=dfba977e-1945-4976-b00a-24be9908e3cd%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-ATYtnXPE7MOa
* Commit ID: 7378ac1367bcbb601898512e7065d9e064d05929
* Result: FAILED
* [Build Logs](https://i1emq028pj.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=03823b97-3bb8-4f23-9b30-38310d51bc0f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: f5cebfa473128c7122045606a571e9fce59e4085
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a18d8df6-3e9d-4b1f-b1c2-4792936483da%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubStandardCodeBuild8C06-llutOAimTATs
* Commit ID: f5cebfa473128c7122045606a571e9fce59e4085
* Result: SUCCEEDED
* [Build Logs](https://jsv9u8gote.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=573c8447-5cb8-48ce-8cd7-5d4351a42805%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-ATYtnXPE7MOa
* Commit ID: f5cebfa473128c7122045606a571e9fce59e4085
* Result: SUCCEEDED
* [Build Logs](https://i1emq028pj.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8e2ee385-7d55-484d-8a78-d47fced52d03%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 68bc502e3f3642bfced0903d715cfa72e0eb6a1d
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=252dbd1d-117e-4a39-ae01-85e3dd069151%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-ATYtnXPE7MOa
* Commit ID: 68bc502e3f3642bfced0903d715cfa72e0eb6a1d
* Result: SUCCEEDED
* [Build Logs](https://i1emq028pj.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=622a37db-69ef-4c63-b04c-4672c73cfd78%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: awswrangler/_distributed.py [hunk 1]
```diff
@@ -25,6 +25,9 @@ class MemoryFormatEnum(Enum):
     PANDAS = "pandas"
 
 
+FunctionType = TypeVar("FunctionType", bound=Callable[..., Any])
```
Comment (kukushking) [inline awswrangler/_distributed.py:28]: Nice!

**Review by kukushking — APPROVED** [review]

**Comment by kukushking** [comment]
2 cents: maybe it's worth to have a separate test case for this.

**Review by malachi-constant — APPROVED** [review]

**Pull request merged**